src/gemm_v2.cpp

```
@@ -324,6 +326,173 @@ inline void ProfilingRecordStop(const Handle& handle, HipEventPtr& start, HipEve
     handle.ResetKernelTime();
     handle.AccumKernelTime(mS);
 }
+
+#define HIP_BLAS_STATUS_CHECK(cmd)                       \
+    do                                                   \
+    {                                                    \
+        hipblasStatus_t e = cmd;                         \
+        if(e != hipblasStatus_t::HIPBLAS_STATUS_SUCCESS) \
+        {                                                \
+            return e;                                    \
+        }                                                \
+    } while(0)
```

I would throw an exception and make it a function instead of define.
Moreover the exception has already been thrown, but for no reasons it happened few layer later, at line 802, 1096 and 1388. Those checks and the status propagating can be removed as well.

---

Thoughts on the updated version @CAHEK7?

I made the changes, but still required a macro to make it easy to call the templated method that throws when status isn't success. I think this was the cleanest way to do this, but curious if you had a suggestion that would be better.

Thanks!

---

It can be something like this:
```C++
static void hipblas_status_check(hipblasStatus_t status)
{
    if(status != hipblasStatus_t::HIPBLAS_STATUS_SUCCESS)
    {
        MIOPEN_THROW(miopenStatusInternalError, "hipBlasLt error encountered");
    }
}

hipblas_status_check(gemm.setProblem(...));
```
THere is no need to pass whole function into the checker, we need just a status.

---

hehe good point! 

I will update. Thanks!